allow switching to modern typeface alternatives with -moderntypeface arg

River-Salmon opened the pull request:
I've generated bitmap fonts from more legible fonts and added a commandline argument to enable them instead of the typical display fonts. 
BigUpper, BigFont, SmallFont are using lexend. The HUD font is using MomoTrustDisplay. 
I've included the relevant OFL license file in the wadsrc. 
Currently only implemented for Doom. 
<img width="435" height="190" alt="image" src="https://github.com/user-attachments/assets/b840330e-6940-47e0-886d-5978c0b3fdff" />
<img width="1004" height="889" alt="image" src="https://github.com/user-attachments/assets/0b467383-3c73-4eb1-b36c-65d32d738c96" />

River-Salmon committed:
allow switching to modern typeface alternatives with -moderntypeface arg

andOlga commented:
- Why are you including a jizzilion copies of the same "undefined" glyph here? Why is it present even in places where actual characters should be? The entire cyrillic range is just this box, for example. This will break languages.

- Why is this a CLI argument and not a CVAR? This should go to the accessibility menu, in my opinion.

RicardoLuis0 commented:
> * Why is this a CLI argument and not a CVAR? This should go to the accessibility menu, in my opinion.

i'm sure it _will_ be added as a cvar, it's just a draft for now

River-Salmon committed:
Composite of Lexend and Noto Sans:

-Noto sans symbols are now included for Small and BigUpper to cover Greek and Cyrillic writing systems.
-Because the "modern typeface" is now no longer one single font, I've given the source folders more generic names.

River-Salmon commented:
This is probably being shelved in favor of more robust font rendering support after the update to c++20.
